app/scripts/app/models/variable_exploration/available_methods.component.js

```diff
@@ -0,0 +1,98 @@
+"use strict";
+angular
+  .module("chuvApp.models")
+  .component("availableMethods", {
+    bindings: {
+      getFocusedVariable: '<',
+      zoom: '='
+    },
+    require: {
+    },
+    controllerAs: "avmethods",
+    controller:[
+      'Variable',
+      "MLUtils",
+      function(Variable, MLUtils){
```

Comment:
With the components architecture+es6, the use of "vm" is obsolete imo. you put stuff on "this" and it will be available in the templates by default as "$ctrl". If the fat arrow notation is used instead of "functino(){}", then there is no risk of surprises anymore as of what "this" really is.